docs/concepts/projects.md

```diff
@@ -39,6 +42,61 @@ dependency list from the command line with `uv add` and `uv remove`. uv also sup
 
     See the official [`pyproject.toml` guide](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/) for more details on getting started with a `pyproject.toml`.
 
+## Creating projects
+
+uv supports initializing a project with `uv init`. By default, uv will create a project in the
+working directory. Projects can be created in a target directory by providing a name, e.g.,
+`uv init foo`. If there's already a project in the target directory, i.e., there's a
+`pyproject.toml`, uv will exit with an error.
+
+By default, uv will create a project for an [application](#applications). However, the `--lib` flag
+can be used to create a project for a [library](#libaries).
+
+## Project types
+
+uv supports two major types of projects, **applications** and **libaries**.
+
+### Applications
+
+Application projects are suitable for for web servers, scripts, and command-line interfaces.
+
+Application projects have the following traits:
+
+- A build backend is not required
+- Source code is often at the top-level, e.g., `hello.py`
+- The project package is not installed in the project environment by default
+
+By default, an application is not intended to be built and distributed as a Python package. However,
+the `--package` flag can be passed to `uv init` to create an application that is distributable,
+e.g., if you want to distribute a command-line interface via PyPI. When requested, uv will define a
+build backend for the project and install it into the project environment
+
+Similarly, a build backend can be manually defined to create a distributable package for an
+application. This may require changes to the project directory stucture, depending on the build
+backend. Once a build backend is defined, uv will attempt to install the project package into the
+project environment.
+
+### Libaries
+
+A library is a project that is intended to be built and distributed as a Python package. A library
+provides functions and objects for other projects to consume.
+
+Library projects have the following traits:
+
+- A build backend is required
+- Source code is typically in `src/{package}/...` directories
+- The project package is installed in the project environment
```

Comment:
Should we say "virtual environment" here?

Reply:
I generally try to talk about the "project environment" or, if ambiguous, "project virtual environment". Just to move the virtual environment concept further from the users.